RFC Draft.md

```diff
@@ -0,0 +1,583 @@
+# RFC: Amplify JS TypeScript Improvements
+
+Amplify JS is looking to improve our TypeScript support across the library to better serve our customers and provide a more intuitive & idiomatic developer experience. To this end, we are requesting your feedback on a variety of changes and improvements that will be available to you in our next major version release.
+
+This RFC is broken down into sections covering:
+
+- Library-wide TypeScript improvements
+- Changes to some of our core utilities such as `Hub` and Amplify configuration
+- Specific improvements to the `Auth`, `Storage`, and `API` categories and associated APIs
+
+We're also requesting feedback on any other TypeScript issues or pain points that you may have encountered not explicitly covered in this RFC.
+
+# Library-wide TypeScript Improvements
+
+Amplify JS will be making the following improvements to our TypeScript support. These improvements will be applied across the entire library, not just the categories highlighted below.
+
+- `strict` typings — We will be applying `strict` mode to the entire library to improve the usability of our types.
+- Better runtime error typing — We will provide utilities for asserting type information of runtime errors emitted by Amplify.
+- Upgraded TypeScript version — We will be upgrading the version of TypeScript that Amplify uses and provide explicit type definitions for customers using older versions. This will provide a variety benefits such as removing the need to specify `skipLibCheck` when using Amplify with newer versions of TypeScript.
+
+# Utility Changes
+
+Amplify is proposing the following changes to our core utilities.
+
+## TypeScript support for Amplify Hub channels
+
+We are improving developer experience by adding strict type support to Hub channels, events, and payloads. An example of the developer experience when listening for `auth` events is highlighted below.
+
+**Amplify v5 & v6 (`aws-amplify@^5`)**
```

Comment:
Do we need that ` & v6`?

Reply:
Should we also mention v5 is current version, it could be confusing if a customer doesn't know the current version or at least they would have to check that

Reply:
Kinda fair actually .. why not just refer to everything as *current* vs *proposed*?